### ISSUE TITLE
private post options

### ISSUE THREAD

**Issue by hoodiek** [glitch-soc/mastodon#2]
private post federation is a powerful feature, but there's a lot of other details that can be worked out alongside it and improved.

exposing more options to users to allow them to customize their experience and capabilities is, in my opinion always the better route for devs to take. for private posts and federation, there are many things that can/should be exposed to the user.

an example would be fine-tuning on federation, IE, levels of trust in your followers, for those that end up with a massive account.

the first set of options to start with is as follows:
1. distribute to all followers
this option is pretty simple, and trusts all your followers with your private posts without any checks or balances.
2. distribute to followers on mastodon instances
this option allows users to trust based on technology, by allowing the user to know that their toots will only ever be boosted by a hacked/modified mastodon instance that does not respect post privacy. this is also a fairly good middleground, and would be where i set the default for this set of settings
3. distribute to people who follow me that i also follow (mutuals)
this is another way to limit your private posts that would be very powerful for users to have, by letting you only trust mutuals, it also enforces a good balance for more private posts such as venting and the like, it allows you to only let people who care about you, and you care about them (by care i mean follow) see your private and thus most vulnerable posts
4. distribute to only users on my instance (possibly allow another subsection choice on only mutuals

i will update this post with any other ideas i have to flesh out and improve this feature as they come to me.

EDIT:
the description in the drop down menu should change appropriately (and possibly have a gear you can click to jump to that part of the settings menu) for private posts

**Comment by deli73** [comment]
Yeah I think it'd be good to have all of these options tbh, though personally I'd have mutuals be the default

**Comment by hoodiek** [comment]
that's a good point, but i think the biggest thing (that i left out) is that the description in the drop down menu should change appropriately (and possibly have a gear you can click to jump to that part of the settings menu)

**Comment by wolfteeth** [comment]
In practice the privacy options I want:
- Mutuals, local and remote
- Local followers only (don't federate)
- Custom privacy filters - I add users to a list. If they follow me, they receive these private posts.

I'm not sure "all followers" has a use case - if I'm OK with my post flying out to all followers in the fediverse where it may be boosted on instances that don't respect privacy, I'd probably just post "Unlisted". I'd be worried about how GNUsocial and other non-Mastodon instances would deal with private posts in general - does only the follower see it, or does it appear in a public federated timeline?

"Only Mastodon" could be good but I'm not sure how to tell. For greater interoperability, I wonder if the OStatus standard could be updated to have a way to find out if an instance respects privacy - default would be no if the instance software hasn't implemented a response - so if an instance claims to respect privacy, private posts would be allowed to federate to it.

Just some ideas. I'd kind of like to see federated deletion hashed out before private post federation. Private posts are the things I'd be most concerned about getting out into the wild.

**Comment by Gingeh** [comment]
Any update on this in 2023? I don't really feel comfortable using followers-only as a privacy option and would very much prefer to use a mutuals-only setting.
